src/dl.cpp

```diff
@@ -68,7 +70,25 @@ dl_error_t dl_instance_load( dl_ctx_t             dl_ctx,          dl_typeid_t
 	DL_ASSERT( (uint8_t*)instance + header->instance_size > packed_instance || (uint8_t*)instance < packed_instance + header->instance_size );
 	memcpy( instance, packed_instance + sizeof(dl_data_header), header->instance_size );
 
-	dl_internal_patch_instance( dl_ctx, root_type, (uint8_t*)instance, 0x0, (uintptr_t)instance );
+	if( header->version == 1 )
+		dl_internal_patch_instance( dl_ctx, root_type, (uint8_t*)instance, 0x0, (uintptr_t)instance );
+	else if( header->need_slow_patching )
+		dl_internal_patch_instance( dl_ctx, root_type, (uint8_t*)instance, 0x0, (uintptr_t)instance - sizeof( dl_data_header ) );
+	else
+	{
+		uintptr_t offset_shift      = sizeof( uintptr_t ) * 4;
+		uintptr_t offset_mask       = ( 1ULL << offset_shift ) - 1;
+		uintptr_t base_ptr          = (uintptr_t)instance - sizeof(dl_data_header);
+		uint8_t* patch_mem          = (uint8_t*)base_ptr;
+		uintptr_t offset_to_pointer = header->first_pointer_to_patch; // 0 is the patch terminator
+		while( offset_to_pointer != 0 )
+		{
+			patch_mem += offset_to_pointer;
+			uintptr_t offsets      = *(uintptr_t*)patch_mem;
+			offset_to_pointer      = offsets >> offset_shift;
```

Comment:
This loop might need a check for offset being outside the instance and return an error... otherwise this is a potential security issue where someone would be able to write whatever they want in memory :) That will be, correctly, reported by coverity.

Reply:
Cool, I'm looking forward to be invited to the coverity project.

Reply:
You already are :)

Reply:
Great. I just didn't get the memo.